cli/lib/kontena/cli/stacks/yaml/opto/service_link_resolver.rb

```diff
@@ -12,6 +12,7 @@ def resolve
         services = client.get("grids/#{current_grid}/services")['services']
         services = filter_by_image(services, image_filter) if image_filter
         services = filter_by_name(services, name_filter) if name_filter
+        raise "No service matched the given filter(s)" if services.size == 0
```

Comment:
I thought that by raising it would be obvious for the user that the match didn't match anything. But now thinking harder it does not make sense as the variable can be optional and in that case raising would be bad. I'll re-work on this.